google/cloud/bigquery/dbapi/_helpers.py

```diff
@@ -189,6 +189,10 @@ def bigquery_scalar_type(value):
     elif isinstance(value, numbers.Real):
         return "FLOAT64"
     elif isinstance(value, decimal.Decimal):
+        if pyarrow is None:
+            # We don't try to manually guess if it's maybe a BIGNUMERIC.
+            return "NUMERIC"
```

Comment:
Just to confirm, we indeed don't want to inspect the number of digits and try to deduce whether the type is actually BIGNUMERIC? As that would essentially mean duplicating pyarrow's logic?

Reply:
Actually, I double-checked the docs, and there is a `Decimial.as_tuple()` method that would be very useful for this.

```python
vtuple = value.as_tuple()
# NUMERIC values have precision of 38 (number of digits)
# and scale of 9 (number of fractional digits), so there can
# be at most 38 - 9 = 29 whole number digits.
# https://cloud.google.com/bigquery/docs/reference/standard-sql/data-types#decimal_types
if vtuple.digits <= 38 and vtuple.exponent <= 29:
    return "NUMERIC"
else:
    return "BIGNUMERIC"
```

Reply:
Actually, I double-checked the docs, and there is a [`Decimial.as_tuple()` method](https://docs.python.org/3/library/decimal.html#decimal.Decimal.as_tuple) that would be very useful for this.

```python
vtuple = value.as_tuple()
# NUMERIC values have precision of 38 (number of digits)
# and scale of 9 (number of fractional digits), so there can
# be at most 38 - 9 = 29 whole number digits.
# https://cloud.google.com/bigquery/docs/reference/standard-sql/data-types#decimal_types
if vtuple.digits <= 38 and vtuple.exponent <= 29:
    return "NUMERIC"
else:
    return "BIGNUMERIC"
```

Reply:
That's precisely the Decimal method that caught my eyes, thus I asked. :+1: 

Yes, let's do that.

Reply:
Following the offline debate and posting here for the reference - interpreting the exponent might be somewhat trickier:

```py
>>> Decimal("1.0E+30").as_tuple()
DecimalTuple(sign=0, digits=(1, 0), exponent=29)
>>> Decimal("1E+30").as_tuple()
DecimalTuple(sign=0, digits=(1,), exponent=30)
```
The same number in both cases, but with different exponents. It does not fit into backend's `NUMERIC` (`BIGNUMERIC` should be used), but checking if `exponent <= 29` is not sufficient by itself.